src/main/java/thecodex6824/thaumicaugmentation/api/impetus/node/prefab/ImpetusNode.java

```diff
@@ -149,219 +149,252 @@ public boolean removeOutput(IImpetusNode output) {
         return true;
     }
-    
+
     @Override
     public double getMaxConnectDistance(IImpetusNode toConnect) {
         return 8.0;
     }
-    
+
     @Override
     public boolean addInputLocation(DimensionalBlockPos toConnect) {
         if (inputs.size() == maxInputs && !inputs.contains(toConnect))
             throw new IndexOutOfBoundsException("Exceeded maximum amount of inputs for node (" + inputs.size() + ")");
-        
+
         return inputs.add(toConnect);
     }
-    
+
     @Override
     public boolean addOutputLocation(DimensionalBlockPos toConnect) {
         if (outputs.size() == maxOutputs && !outputs.contains(toConnect))
             throw new IndexOutOfBoundsException("Exceeded maximum amount of outputs for node (" + outputs.size() + ")");
-        
+
         return outputs.add(toConnect);
     }
-    
+
     @Override
     public boolean removeInputLocation(DimensionalBlockPos toRemove) {
         return inputs.remove(toRemove);
     }
-    
+
     @Override
     public boolean removeOutputLocation(DimensionalBlockPos toRemove) {
         return outputs.remove(toRemove);
     }
-    
+
     @Override
     public Set<DimensionalBlockPos> getInputLocations() {
         return inputs;
     }
-    
+
     @Override
     public Set<DimensionalBlockPos> getOutputLocations() {
         return outputs;
     }
-    
+
     @Override
     public Set<IImpetusNode> getInputs() {
         return inputs.stream().map(loc -> graph.findNodeByPosition(loc)).filter(Objects::nonNull).collect(Collectors.toSet());
     }
-    
+
     @Override
     public Set<IImpetusNode> getOutputs() {
         return outputs.stream().map(loc -> graph.findNodeByPosition(loc)).filter(Objects::nonNull).collect(Collectors.toSet());
     }
-    
+
     @Override
     public DimensionalBlockPos getLocation() {
         return loc;
     }
-    
+
     @Override
     public void setLocation(DimensionalBlockPos location) {
         if (!loc.isInvalid())
             graph.removeNode(this);
-        
+
         loc = location;
         graph.addNode(this);
     }
-    
+
     @Override
     public Vec3d getBeamEndpoint() {
         return new Vec3d(loc.getPos().getX() + 0.5, loc.getPos().getY() + 0.5, loc.getPos().getZ() + 0.5);
     }
-    
+
     @Override
     public boolean shouldPhysicalBeamLinkTo(IImpetusNode other) {
         return true;
     }
-    
+
     @Override
     public boolean shouldEnforceBeamLimitsWith(IImpetusNode other) {
         return true;
     }
-    
+
     @Override
     public void unload() {
         graph.removeNode(this);
     }
-    
+
     @Override
     public void destroy() {
         for (IImpetusNode node : graph.getInputs(this))
             removeInput(node);
-            
+
         for (IImpetusNode node : graph.getOutputs(this))
             removeOutput(node);
-        
+
         unload();
     }
-    
-    protected void initServer() {
+
+    @Override
+    public boolean hasUnloadedNodes(World world) {
```

Comment:
This method seems to do more than just returning if there are unloaded nodes - you can have a function returning `hasUnloadedNodes`, but maybe split the logic for actually checking / trying to load them into a `tryConnectUnloadedNodes` method or something like that.

Alternatively, you could put that logic into `NodeHelper` (substituting field access with the getters). I'm not sure which one would be a better place for it.

Reply:
I think that this is a better place as if I would move it I would need to add more methods to the interface, and those methods wouldn't be all that useful.